packages/nextjs/components/batch/BuildersGrid.tsx

```diff
@@ -0,0 +1,80 @@
+import { useEffect, useState } from "react";
+import Link from "next/link";
+import { Address } from "~~/components/scaffold-eth";
+
+interface Builder {
+  id: number;
+  address: string;
+}
+
+// This needs to be in a server component or API route
+// async function getBuilderAddresses() {
+//   const buildersPath = path.join(process.cwd(), 'app/builders');
+//   try {
+//     const directories = await readdir(buildersPath, { withFileTypes: true });
+//     return directories
+//       .filter(dirent => dirent.isDirectory())
+//       .map(dirent => dirent.name)
+//       .filter(name => name.startsWith('0x'));
+//   } catch (error) {
+//     console.error('Error reading builders directory:', error);
+//     return [];
+//   }
+// }
+
+// Client component
+const BuildersGrid = () => {
+  const [builders, setBuilders] = useState<Builder[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    fetch("/api/builders")
+      .then(res => res.json())
+      .then(addresses => {
+        const buildersList = addresses.map((address: string, index: number) => ({
+          id: index,
+          address: address,
+        }));
+        setBuilders(buildersList);
+        setIsLoading(false);
+      })
+      .catch(error => {
+        console.error("Error fetching builders:", error);
+        setIsLoading(false);
+      });
+  }, []);
```

Comment:
This is a matter of preference, but react-query would fit well here.

Reply:
Also, beside getting the builders by their profile pages, we could get whoever made checkin, even the ones without a profile page. Check this [discussion](https://github.com/BuidlGuidl/batch11.buidlguidl.com/issues/4#issuecomment-2485640200).

Reply:
Can we get the details from the builders that checked-in?
Okay, I will try it out and see, but not sure how to figure it, but I will definitely try all my best. Thanks so much

Reply:
@gboigwe, this one is still open. We want to display all checked-in builders, not just those with a personal profile.